build/tasks/EmbedAppNameInHost.cs

```diff
@@ -0,0 +1,175 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Text;
+
+namespace Microsoft.DotNet.Cli.Utils
+{
+    /// <summary>
+    /// Embeds the App Name into the AppHost.exe
+    /// </summary>
+    public class EmbedAppNameInHost : Build.Utilities.Task
```

Comment:
In general, we are marking everything internal unless it needs to be exposed for extensibility. Is this the case?

Reply:
This is not part of what we ship. 
This is a custom build task that is needed for building the shims for global tool. 
MsBuild tasks need to be public.